### PR TITLE
[ADDED] KeyValue: RePublish option in KeyValueConfiguration

### PR DESCRIPTION
**Pull request by kozlovic**
Ability to create a KV with the re-publish feature.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

**Comment by kozlovic** [comment]
@bruth Wanted to add you as a reviewer, but for some reason I could not find you in the list. Maybe you need to be added into some GH lists.

**Review by codegangsta — APPROVED** [review]
Nice and easy. LGTM 👍

File: kv.go [hunk 1]
```diff
@@ -401,6 +402,7 @@ func (js *js) CreateKeyValue(cfg *KeyValueConfig) (KeyValue, error) {
 		MaxMsgs:           -1,
 		MaxConsumers:      -1,
 		AllowDirect:       true,
+		RePublish:         cfg.RePublish,
```
Comment (kozlovic) [inline kv.go:405]: Humm... I did not test the situation where we call that on an existing stream. We may then have the issue I was discussing during the call at line 425 (where we compare the config that we got from a lookup vs what we are trying to set...). Will see if I can add a test to simulate the upgrade.

Comment (kozlovic) [reply kv.go:405]: Well, maybe that's ok then that the CreateKeyValue fails, the user should then administratively update the stream (if at all possible, not sure if we support updating this). Still will add a test.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by kozlovic** [comment]
@codegangsta @derekcollison I think that the stream cannot be updated with RePublish, meaning it may not reject the update per-se (as it did with pull consumer MaxWaiting), but it is actually not *updating* the stream and therefore the feature is not enabled. For this PR maybe I keep it that it would fail if the stream is originally not created with RePublish (don't try to auto-update). Will then need to add a test in server to demonstrate the issue, or @derekcollison if you think that it should be supported or should work, I may have done something wrong in my client test...

**Comment by derekcollison** [comment]
I am ok with rejecting, just need to formalize the rejection in the server code. If need be we can add update support at a later time.

**Comment by kozlovic** [comment]
Ok, so will update this PR with test that verifies that 2nd create with republish fails as expected, then will go to server to add test that ensure that we reject update properly (started to work on a test and update is not rejected, but republish update is clearly not applied).

**Comment by derekcollison** [comment]
Perfect thanks.

**Pull request merged**